---
description: '詳細情報: C6310'
title: C6310
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C6310
helpviewer_keywords:
- C6310
ms.assetid: e759eb63-883f-4c3e-bf2f-b924ff511405
ms.openlocfilehash: 9899acffb6e1035a937de46eef6a5c1bbc9ceabe
ms.sourcegitcommit: d6af41e42699628c3e2e6063ec7b03931a49a098
ms.translationtype: MT
ms.contentlocale: ja-JP
ms.lasthandoff: 12/11/2020
ms.locfileid: "97284227"
---
# <a name="c6310"></a>C6310

> 警告 C6310: 例外フィルター内の無効な定数は、予期しない動作を発生させる原因となる可能性があります

このメッセージは、構造化例外ハンドラのフィルター式で無効な定数が検出されたことを示します。 構造化例外ハンドラーのフィルター式で使用するように定義された定数は次のとおりです。

- `EXCEPTION_CONTINUE_EXECUTION`

- `EXCEPTION_CONTINUE_SEARCH`

- `EXCEPTION_EXECUTE_HANDLER`

これらの値は、ランタイムヘッダーファイル excpt.h で定義されています。

前の一覧にない定数を使用すると、予期しない動作が発生する可能性があります。

## <a name="example"></a>例

この警告が発生するコード例を次に示します。

```cpp
#include <excpt.h>
#include <stdio.h>
#include <windows.h>

BOOL LimitExceeded();

void fd( )
{
  __try
  {
    if (LimitExceeded())
    {
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);
    }
    else
    {
      // code
    }
  }
  __except (EXCEPTION_ACCESS_VIOLATION)
  {
    puts("Exception Occurred");
  }
}
```

この警告を解決するには、次のコードを使用します。

```cpp
#include <excpt.h>
#include <stdio.h>
#include <windows.h>

BOOL LimitExceeded();

void fd( )
{
  __try
  {
    if (LimitExceeded())
    {
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);
    }
    else
    {
      // code
    }
  }
  __except (GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)
  {
    puts("Exception Occurred");
  }
}
```
